HIRS_Utils/src/main/java/hirs/tpm/eventlog/uefi/UefiGuid.java

```diff
@@ -0,0 +1,415 @@
+ case "53746f72-6167-11aa-aa11-00306543ecac":reference = "Apple Core Storage Container"; break;
+ case "6a898cc3-1dd2-11b2-99a6-080020736631":reference = "ZFS Partition"; break;
+
+ // Chrome OS GUIDS
+ case "2568845d-2332-4675-bc39-8fa5a4748d15":reference = "Chrome OS kernel "; break;
+ case "3cb8e202-3b7e-47dd-8a3c-7ff2a13cfcec":reference = "Chrome OS rootfs "; break;
+ case "2e0a753d-9e48-43b0-8337-b15192cb1b5e":reference = "Chrome OS future use "; break;
+
+ // Android GUIDS
+ case "fe3a2a5d-4f32-41a7-b725-accc3285a309":reference = "Android Bootloader"; break;
+ case "114eaffe-1552-4022-b26e-9b053604cf84":reference = "Android Bootloader 2"; break;
+ case "49a4d17f-93a3-45c1-a0de-f50b2ebe2599":reference = "Android Boot"; break;
+ case "4177c722-9e92-4aab-8644-43502bfd5506":reference = "Android Recovery"; break;
+ case "38f428e6-d326-425d-9140-6e0ea133647c":reference = "Android System"; break;
+ case "bd59408b-4514-490d-bf12-9878d963f378":reference = "Android Config"; break;
+ case "8f68cc74-c5e5-48da-be91-a0c8c15e9c80":reference = "Android Factory"; break;
+ case "ac6d7924-eb71-4df8-b48d-e267b27148ff":reference = "Android OEM"; break;
+
+ // MISC GUIDs
+ case "5023b95c-db26-429b-a648-bd47664c8012":reference = "Built-in EFI Shell"; break;
+ case "610a0202-d308-00c4-0000-000004300d06":reference = "Mystery UUID"; break;
+ case "00000000-0000-0000-0000-000000000000":reference = "Empty UUID"; break;
+
+ default: reference = "Unknown GUID reference";
+  }
+ return reference;
+ }
+
+/**
+ * Returns a string of the entity that the UUID represents.
+ * Contains a Vendor String lookup on the UUID.
+ * @return UUID description.
+ */
+public String toString() {
+   String guidinfo = "";
+   guidinfo = uuid.toString() + " : " + getVendorTableReference();
+   return guidinfo;
+ }
+
+/**
+ * Returns a string of the entity that the UUID represents.
+ * Does not contain a vendor lookup on the UUID.
+ * @return UUID description.
+ */
+public String toStringNoLookup() {
+ return  uuid.toString();
+}
+/**
+ * Returns a string of the entity that the UUID represents.
+ * Does not contain a vendor lookup on the UUID.
+ * @param guid byte array holding the guid data.
+ * @return true if the UUID has a valid structure.
+ */
+public static boolean isValidUUID(final byte[] guid) {
+ boolean valid = false;
+ UUID tmpUuid = processGuid(guid);
+ if (tmpUuid.toString().length() != 0)  {
+       valid = true;
+      }
+    return (valid);
+ }
+
+/**
+ * Checks to see if the uuid is the test or Empty UUID ("00000000-0000-0000-0000-000000000000").
+ * @return true if the uuid is the Empty UUID, false if not
+ */
+public boolean isEmptyUUID() {
+ return uuid.toString().equals("00000000-0000-0000-0000-000000000000");
+ }
+
+/**
+ * Checks to see if the uuid is the Empty UUID or an unknown.
+ * @return true if the uuid is the Empty UUID, false if not
+ */
+public boolean isUnknownUUID() {
+   if (getVendorTableReference().equals("Unknown GUID reference")) {
```

Comment:
You can just replace the entire if statement by making it a return statement like you did on line 393

Reply:
OK